ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/helpers/BeaconStateAccessors.java

```diff
@@ -90,6 +91,22 @@ public Optional<BLSPublicKey> getValidatorPubKey(BeaconState state, UInt64 valid
                 }));
   }
 
+  public boolean validateWithdrawalCredentials(
+      BeaconState state, BlsToExecutionChange addressChange) {
+    final int validatorIndex = addressChange.getValidatorIndex().intValue();
+    if (state.getValidators().size() <= validatorIndex || validatorIndex < 0) {
+      return false;
+    }
+    final Bytes32 withdrawalCredentials =
+        state.getValidators().get(validatorIndex).getWithdrawalCredentials();
+    if (withdrawalCredentials.get(0) != config.getBlsWithdrawalPrefix().get(0)
+        || withdrawalCredentials.slice(1, 31)
```

Comment:
Should be able to just use `.slice(1)` here to match what we're doing to the sha256.